### PR TITLE
Remove DeclarativeCallAgent from stable API

### PR DESCRIPTION
**Pull request by prprabhu-ms**
This is new API needed for a beta-only feature. Let's remove it from stable builds for now.

### CODE REVIEW

**Review by dmceachernmsft — APPROVED** [review]

**Review by PorterNan — APPROVED** [review]

File: packages/calling-stateful-client/src/StatefulCallClient.ts [hunk 1]
```diff
@@ -128,6 +128,7 @@ export interface StatefulCallClient extends CallClient {
     stream: LocalVideoStreamState | RemoteVideoStreamState
   ): void;
 
+  /** @conditional-compile-remove(one-to-n-calling) */
```
Comment (anjulgarg) [inline packages/calling-stateful-client/src/StatefulCallClient.ts:131]: Why is this added here?
